### DISCUSSION TITLE
Updating Helix

### DISCUSSION THREAD

**Discussion by rcortezk9** [helix-editor/helix#6507] [Q&A]
I may be asking a dumb question. I installed Helix on Linux from source (if my memory serves me right) and I'm not sure how I would update Helix.

**Comment by the-mikedavis** [comment]
You can `git pull` in your local copy of the Helix repository and re-run 
`cargo install --locked --path helix-term`

**Reply by pickfire** [reply]
Or another alternative is to install those `-git` from your package manager if it have one, like arch linux. Which works well with topgrade https://github.com/topgrade-rs/topgrade.

**Comment by rcortezk9** [comment]
I think this cause to have it installed twice with caused some issue with the themes. I uninstalled it.

**Reply by cd-a** [reply]
Be sure to follow this step here when installing from source: https://docs.helix-editor.com/install.html#configuring-helixs-runtime-files

**Comment by cd-a** [comment]
Going forward, you should decide what channel you want to be on, release or master.

If you're good with releases, uninstall the one you've built from source and use your distro's package manager to install helix, that should keep it up to date.

If you want to be on the bleeding edge or update whenever it suits you, see @the-mikedavis comment. Note that this will never prompt you to update, so you will have to remember to once in a while re-compile or set a reminder :)

**Comment by ardrigh** [comment]
Since I already had a git clone of the repo, it was a matter of creating a branch from the version tag, then running the compile command again.

```shell
> git fetch --all
> git checkout tags/23.03 -b 23.03
> cargo install --path helix-term --locked
...
> hx --version
helix 23.03 (3cf03723)
> which hx
/home/user/.cargo/bin/hx
```

If you don't want to use the version releases, you would just `git rebase` the `master` branch and compile that instead of the version tag branch
